packages/backend/src/models/table-metadata.ts

```diff
@@ -53,6 +55,22 @@ class TableMetadata extends Base {
     }
     return true
   }
+
+  async stripInvalidKeys(rows: ITableRow[]): Promise<ITableRow[]> {
+    const columns = await this.$relatedQuery('columns')
+    const columnIdsSet = new Set(columns.map((column) => column.id))
+
+    return rows.map((row) => {
+      const validKeys = Object.keys(row.data).filter((key) =>
+        columnIdsSet.has(key),
+      )
+      const validData = validKeys.reduce(
+        (acc, key) => ({ ...acc, [key]: row.data[key] }),
+        {},
+      )
+      return { ...row, data: validData }
+    })
+  }
```

Comment:
nit: if you want to make this compact:

```suggestion
    return rows.map((row) => ({
      ...row,
      data: pickBy(row.data, (_, key) => columnIdsSet.has(key)),
    }))
  }
```